src/variant_detection/variant_output.cpp

```diff
@@ -94,24 +153,36 @@ void find_and_output_variants(std::map<std::string, int32_t> & references_length
     }
 }
 
-//!\overload
 void find_and_output_variants(std::map<std::string, int32_t> & references_lengths,
                               std::vector<Cluster> const & clusters,
                               cmd_arguments const & args,
                               std::filesystem::path const & output_file_path)
 {
+    bio::var_io::header hdr;
+    write_header(references_lengths, args.vcf_sample_name, hdr);
+
+    std::vector<bio::var_io::default_record<>> records;
+    write_records(clusters, args, records);
```

Comment:
I don't know if seed is important, but right now you are creating all records and adding them to a vector. And then you iterator over the vector to write each record. Why not write the records immediately when you create them?